recipes/poppler/all/conanfile.py

```diff
@@ -132,125 +126,119 @@ def _minimum_compilers_version(self):
-        self._cmake.definitions["ENABLE_LIBOPENJPEG"] = "openjpeg2" if self.options.with_openjpeg else "none"
-        if self.options.with_openjpeg:
-            # FIXME: openjpeg's cmake_find_package should provide these variables
-            self._cmake.definitions["OPENJPEG_MAJOR_VERSION"] = tools.Version(self.requires["openjpeg"].ref.version).major
-        self._cmake.definitions["ENABLE_CMS"] = "lcms2" if self.options.with_lcms else "none"
-        self._cmake.definitions["ENABLE_LIBCURL"] = self.options.with_libcurl
-
-        self._cmake.definitions["POPPLER_DATADIR"] = self.deps_user_info["poppler-data"].datadir.replace("\\", "/")
-        self._cmake.definitions["FONT_CONFIGURATION"] = self.options.fontconfiguration
-        self._cmake.definitions["BUILD_CPP_TESTS"] = False
-        self._cmake.definitions["ENABLE_GTK_DOC"] = False
-        self._cmake.definitions["ENABLE_QT5"] = self.options.with_qt and tools.Version(self.deps_cpp_info["qt"].version).major == "5"
-        self._cmake.definitions["ENABLE_QT6"] = self.options.with_qt and tools.Version(self.deps_cpp_info["qt"].version).major == "6"
-
-        self._cmake.definitions["ENABLE_CMS"] = "lcms2" if self.options.with_lcms else "none"
-        self._cmake.definitions["ENABLE_DCTDECODER"] = self._dct_decoder
-        self._cmake.definitions["USE_FLOAT"] = self.options.float
-        self._cmake.definitions["RUN_GPERF_IF_PRESENT"] = False
+            tc.variables["ENABLE_BOOST"] = self.options.splash
+
+        tc.variables["FONT_CONFIGURATION"] = self.options.fontconfiguration
+        tc.variables["WITH_JPEG"] = self.options.with_libjpeg
+        tc.variables["WITH_PNG"] = self.options.with_png
+        tc.variables["WITH_TIFF"] = self.options.with_tiff
+        tc.variables["WITH_NSS3"] = self.options.with_nss
+        tc.variables["WITH_Cairo"] = self.options.with_cairo
+        tc.variables["ENABLE_GLIB"] = self.options.get_safe("with_glib", False)
+        tc.variables["ENABLE_GOBJECT_INTROSPECTION"] = self.options.get_safe("with_gobject_introspection", False)
+        tc.variables["WITH_Iconv"] = self.options.get_safe("with_libiconv")
+        tc.variables["ENABLE_ZLIB"] = self.options.with_zlib
+        tc.variables["ENABLE_LIBOPENJPEG"] = "openjpeg2" if self.options.with_openjpeg else "none"
+        tc.variables["ENABLE_CMS"] = "lcms2" if self.options.with_lcms else "none"
+        tc.variables["ENABLE_LIBCURL"] = self.options.with_libcurl
+
+        tc.variables["POPPLER_DATADIR"] = self.deps_user_info["poppler-data"].datadir.replace("\\", "/")
+        tc.variables["FONT_CONFIGURATION"] = self.options.fontconfiguration
+        tc.variables["BUILD_CPP_TESTS"] = False
+        tc.variables["ENABLE_GTK_DOC"] = False
+        tc.variables["ENABLE_QT5"] = self.options.with_qt and Version(self.deps_cpp_info["qt"].version).major == "5"
+        tc.variables["ENABLE_QT6"] = self.options.with_qt and Version(self.deps_cpp_info["qt"].version).major == "6"
```

Comment:
This should be using the new `self.dependencies` the docs jut got merged https://github.com/conan-io/conan-center-index/blob/master/docs/adding_packages/dependencies.md#verifying-dependencys-version